satpy/readers/hrit_base.py

```diff
@@ -347,13 +348,27 @@ def _read_or_memmap_data(self, shape, dtype):
 
     def read_band(self, key, info):
         """Read the data."""
-        shape = int(np.ceil(self.mda['data_field_length'] / 8.))
-        if self.mda['number_of_bits_per_pixel'] == 16:
-            dtype = '>u2'
-            shape //= 2
-        elif self.mda['number_of_bits_per_pixel'] in [8, 10]:
-            dtype = np.uint8
-        shape = (shape, )
+        input_shape = int(np.ceil(self.mda['data_field_length'] / 8.))
+        bpp = self.mda['number_of_bits_per_pixel']
+        if bpp == 16:
+            input_dtype = '>u2'
+            output_dtype = '>u2'
+            input_shape //= 2
+        elif bpp == 10:
+            input_dtype = np.uint8
+            output_dtype = np.uint16
+        elif bpp == 8:
+            input_dtype = np.uint8
+            output_dtype = np.uint8
+        else:
+            raise ValueError(f"Illegal number of bits per pixel: {bpp}")
+        input_shape = (input_shape, )
+        return da.from_delayed(self._read_data(input_shape, input_dtype),
+                               shape=(self.mda['number_of_lines'], self.mda['number_of_columns']),
+                               dtype=output_dtype)
+
+    @dask.delayed
+    def _read_data(self, shape, dtype):
```

Comment:
I know this is still a draft PR but this method should probably be moved outside the class so it doesn't depend on `self` otherwise dask distributed may not be able to serialize it.

Reply:
Actually, since you're using `fromfile`, could you replace this delayed function with a `map_blocks` with no input array? You could use the optional block info keyword args to determine how much to read from the file. This way reading is chunk-based rather than all at once...or maybe that isn't possible or useful?

Reply:
Good point with self, I will try to move it outside the class. Regarding the chunking/map_blocks: my opinion is that the data is too small for it to be interesting, and thus would add a layer of unnecessary complexity. The biggest segment size I know of is 8MB...

Reply:
I see the AHI HRIT "VIS" file is 231MB, but not sure how much of that is the band data. The other band files are 15MB. I'm not super familiar with HRIT internals so LGTM.